citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -381,4 +427,137 @@ mod tests {
+
+        let (mut sink, mut stream) = framed.split();
+
+        let first_packet = stream.next().await.unwrap()?;
+        info!(target: "citadel", "First packet");
+        let greeter_packet: InternalServiceResponse = bincode2::deserialize(&*first_packet)?;
+
+        info!(target: "citadel", "Greeter packet {greeter_packet:?}");
+
+        if let InternalServiceResponse::ServiceConnectionAccepted { id } = greeter_packet {
+            let register_command = InternalServicePayload::Register {
+                uuid: id,
+                server_addr: server_bind_address,
+                full_name: String::from("John"),
+                username: String::from("john_doe"),
+                proposed_password: String::from("test12345").into_bytes().into(),
+            };
+            send(&mut sink, register_command).await?;
+
+            let second_packet = stream.next().await.unwrap()?;
+            let response_packet: InternalServiceResponse = bincode2::deserialize(&*second_packet)?;
+            if let InternalServiceResponse::RegisterSuccess { id } = response_packet {
+                // now, connect to the server
+                let command = InternalServicePayload::Connect {
+                    // server_addr: server_bind_address,
+                    username: String::from("john_doe"),
+                    password: String::from("test12345").into_bytes().into(),
+                    uuid: id,
+                };
+
+                send(&mut sink, command).await?;
+
+                let next_packet = stream.next().await.unwrap()?;
+                let response_packet: InternalServiceResponse =
+                    bincode2::deserialize(&*next_packet)?;
+                if let InternalServiceResponse::ConnectSuccess { cid } = response_packet {
+                    let serialized_message = bincode2::serialize("hi").unwrap();
+                    let message_command = InternalServicePayload::Message {
+                        uuid: id,
+                        message: serialized_message,
+                        cid,
+                        user_cid: cid,
+                        security_level: SecurityLevel::Standard,
+                    };
+
+                    send(&mut sink, message_command).await?;
+
+                    info!(target:"citadel", "Message sent to sink from client");
+
+                    let next_packet = stream.next().await.unwrap()?;
+                    let response_message_packet: InternalServiceResponse =
+                        bincode2::deserialize(&*next_packet)?;
+                    info!(target: "citadel","{response_message_packet:?}");
+
+                    if let InternalServiceResponse::MessageReceived {
+                        cid,
+                        message: _,
+                        peer_cid: _,
+                    } = response_message_packet
+                    {
+                        info!(target:"citadel", "Message {cid}");
```

Comment:
We need 1 more check inside here: Check that the server tells us that we received a message with contents "pong".